src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.ShopOperation;
+import exception.OperationException;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(ShopOperation shopOperation) {
+        String fruit = shopOperation.getFruit();
+        int quantity = Storage.fruitsStorage.get(fruit);
+        int quantityToBuy = shopOperation.getQuantity();
+        if (quantity < quantityToBuy) {
+            throw new OperationException("Operation is impossible, not enough quantity of: "
+                + fruit);
```

Comment:
Ensure that the `OperationException` class is defined in your project. If not, you need to create this custom exception class to handle operation-related errors appropriately.